src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -103,11 +103,19 @@ public override unsafe int FindFirstCharacterToEncode(char* text, int textLength
 #endif
             Debug.Assert(textLength > 0 && ptr < end);
 
+            // For performance on certain platforms, avoid referencing the static table for every value.
+            ReadOnlySpan<byte> allowListLocal = AllowList;
+
             do
             {
                 Debug.Assert(text <= ptr && ptr < (text + textLength));
 
-                if (NeedsEscaping(*(char*)ptr))
+                char value = *(char*)ptr;
+
+                // NeedsEscaping() is lifted below for perf; verify semantics remain consistent.
```

Comment:
Is this change still necessary after https://github.com/dotnet/runtime/pull/39195?

Based on @BrzVlad's comment (https://github.com/dotnet/runtime/pull/39323#issuecomment-658798816), what are the perf numbers before and after, now?

Reply:
The PR was created after @BrzVlad's's https://github.com/dotnet/runtime/pull/39195.

I double-checked against latest code and the perf results of this PR are basically the same. The S.T.E.W changes are responsible for the serialization gains the the S.T.J changes are responsible for the deserialization gains.